pkg/cmd/roachtest/roachtestutil/disk_stall.go

```diff
@@ -111,6 +117,49 @@ func (s *cgroupDiskStaller) Unstall(ctx context.Context, nodes option.NodeListOp
 	}
 }
 
+// StallForDuration stalls the disk for the specified duration. This avoids creating
+// multiple SSH connections in quick succession, which can cause flakiness.
+func (s *cgroupDiskStaller) StallForDuration(
+	ctx context.Context, nodes option.NodeListOption, duration time.Duration,
+) {
+	maj, min := s.device(nodes)
+	cockroachIOController := filepath.Join("/sys/fs/cgroup/system.slice", SystemInterfaceSystemdUnitName()+".service", "io.max")
+
+	// Shuffle the order of read and write stall initiation.
+	rand.Shuffle(len(s.readOrWrite), func(i, j int) {
+		s.readOrWrite[i], s.readOrWrite[j] = s.readOrWrite[j], s.readOrWrite[i]
+	})
+
+	// Prepare bandwidth limit strings for stall and unstall.
+	var limits []string
+	for _, rw := range s.readOrWrite {
+		limits = append(limits, fmt.Sprintf("%s=4", rw.cgroupV2BandwidthProp()))
+	}
+	stallLimits := strings.Join(limits, " ")
+
+	// Prepare max bandwidth strings for unstall.
+	var maxLimits []string
+	for _, rw := range s.readOrWrite {
+		maxLimits = append(maxLimits, fmt.Sprintf("%s=max", rw.cgroupV2BandwidthProp()))
+	}
+	unstallLimits := strings.Join(maxLimits, " ")
+
+	// Build the command to stall the disk for the specified duration and then unstall it.
+	cmd := fmt.Sprintf(`'
+echo %d:%d %s > %s
+sleep %f
+echo %d:%d %s > %s
+'`,
+		maj, min, stallLimits, cockroachIOController,
+		duration.Seconds(),
+		maj, min, unstallLimits, cockroachIOController)
+
+	err := s.c.RunE(ctx, option.WithNodes(nodes), "sudo", "/bin/bash", "-c", cmd)
+	if err != nil {
+		s.f.L().PrintfCtx(ctx, "error in StallForDuration: %v", err)
```

Comment:
The above command can fail under different failure modes. As before, it can fail due to a (transient) network issue, resulting in an SSH flake. It can also fail if the script errors out for any reason, e.g., output file is not available. In either case, I think you want to fail the test. Note that SSH error would have already been detected (in `remoteSession.Run`) and classified as flake.